### ISSUE TITLE
Using Triaxial data

### ISSUE THREAD

**Issue by stuartgo** [njacobsonlab/Pretrained-Actigraphy-Transformer#1]
Hi,
very interesting paper! I am looking forward to fine tune your model on some trixial data I have.

In your paper you write "We use the average triaxial data per minute to align with earlier uniaxial data.", do you just do a simple average of the x,y and z axis data for each minute when you have triaxial data?

And is Monitor-Independent Movement Summary (MIMS) used or only raw data?

Thanks:)

**Comment by Franklinyeruan** [comment]
Hi, that's exciting!

For our triaxial data, we use the minute level MIMS triaxial value, which is the sum of the MIMS x-, y-, and z-axis values (good catch, I need to correct the typo there since it’s not the average!). 

However, at a subsequent step we use StandardScaler from sklearn to standardize each individual dataset (with respect to itself). As such, whether you sum or average the triaxial data (MIMS or raw) you should get the same output after standardization. 

Hope that helps, and let us know if anything is confusing!

- Franklin

**Comment by stuartgo** [comment]
Thanks!
Everything seems to work now:)
It looks like the model only takes in data for 1 week at a time, but in the code it kind of looks like you can change the input size just by changing the function parameter, maybe it should be more clear that the input size needs to be 10080?
I tried adding a dense layer to fit data for one day at a time, but not sure if that makes sense if the encoder is trained for week long data?

**Comment by Franklinyeruan** [comment]
That’s great to hear, and thank you for the feedback!

Theoretically, our model should handle inputs of any length well (especially those under 10,080 minutes, i.e., a week of actigraphy). For example, even during pretraining, the input to the encoder is shorter than one week (i.e., only 10% of each actigraphy sequence is fed to the encoder).

However, the current implementation in the example notebooks is somewhat rigid. Following that code, you would only be able to fine-tune on a week's worth of minute-level actigraphy data.

For now, a quick workaround would be to pad your one day's worth (1,440 minutes) of actigraphy data to 10,080 minutes by adding a lot of zeros to the end of the sequence. The model will only “pay attention” to the first day—you can verify this using the model explainability function.

We are working on a more flexible way to fine-tune PAT on actigraphy inputs of any length and can let you know once that solution is available!

Hope this helps,
- Franklin

**Comment by Franklinyeruan** [comment]
@stuartgo the issue has been fixed- PAT can now take in variable input lengths. See the updated demo notebook under the section `⭐ How to: Fine-tune PAT + Built-in Model Explainability` in the README for an example! 

- Franklin
